src/hotspot/share/ci/ciMethodBlocks.cpp

```diff
@@ -146,7 +147,9 @@ void ciMethodBlocks::do_analysis() {
       case Bytecodes::_ifnonnull   :
       {
         cur_block->set_control_bci(bci);
-        ciBlock *fall_through = make_block_at(s.next_bci());
+        if (s.next_bci() < limit_bci) {
+          ciBlock *fall_through = make_block_at(s.next_bci());
```

Comment:
I see that we already have this check in place for some usages of `make_block_at`. Could we simply move the checks into that method (and assert `!= NULL` at use sides where this should never happen)? 

If not, can we at least remove the unused local variables? Like so:

```suggestion
          make_block_at(s.next_bci());
```

Reply:
My first attempt was to use a macro MAKE_BLOCK_AT_FALLTHROUGH, but decided to minimize code changes instead, especially since the same pattern is used in the c1 code.

The unused local is used in several places in that function.  It seems to be a way to self-ducument the code without using a comment.  If I remove the unused local in one place, then I should probably do it everywhere and add more comments.

Reply:
Okay, thanks for the background, looks good to me as is.